docs/source/developer/tests.rst

```diff
@@ -117,27 +126,170 @@ test is run while `record_flight_data` is in place.
 When the test is completed, change to using `replay_flight_data`:
 
   .. code-block:: python
-    :emphasize-lines: 2,3
+    :emphasize-lines: 2
 
-    def test_example(self):
-        session_factory = self.replay_flight_data(
-            'test_example')
+    def test_example(self, test):
+        session_factory = test.replay_flight_data('test_example')
 
         ...
 
 Now when the test is run it will use the data previously recorded and will not
-contact AWS.  When committing your test, don't forget to include the 
+contact AWS.  When committing your test, don't forget to include the
 `tests/data/placebo/test_example` directory!
 
 Note: if it's necessary to delay CLI calls due to delays in the time it takes
 for an attribute on a resource to be reflected in an API call or any other reason,
-use ``self.recording`` to only sleep when recording json like so:
+use ``test.recording`` to only sleep when recording json like so:
 
   .. code-block:: python
 
     import time
 
     ...
 
-    if self.recording:
+    if test.recording:
       time.sleep(10)
+
+Writing Terraform Bootstrapped Tests for AWS Resources
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Note: These instructions are only for recording data in AWS. For Azure, GCP, or
+other cloud providers, see the corresponding documentation for information on how
+to record data there.
+
+The `pytest-terraform <https://github.com/cloud-custodian/pytest-terraform>`_ plugin
+is used to streamline the creation and removal of AWS resources using Terraform during
+functional testing.
+
+Terraform modules should be organized in `tests/terraform`. So for a test named
+`test_file_example` the terraform files for that would exist in `tests/terraform/file_example`.
+
+Here's an example terraform file for the upcoming code example. It's placed in
+`tests/terraform/file_example/main.tf`.
+
+  .. code-block:: terraform
+
+    resource "local_file" "bar" {
+       content = "bar!"
+       filename = "${path.module}/bar.txt"
+    }
+
+When invoked, this terraform module will create a file `bar.txt` with the
+contents ``bar!``.
+
+When first creating a test, explicitly set the `replay` parameter to ``False``.
+This will force terraform to run on each invocation of the test and perform
+the flight recording function.
+
+Given the above example terraform file, the following code example is used to
+access data from the terraform decorator, passed in as the same name as the
+first parameter, in this case `file_example`
```

Comment:
this is missing the caveat wrt to unit test class derivatives, and using the `test` fixture for access to our existing c7n test api.